networkx/algorithms/summarization.py

```diff
@@ -0,0 +1,187 @@
+"""
+Algorithms for reducing the number of nodes/edges in a graph
+
+*dedensification* reduces the number of edges in unweighted graphs by adding
+compressor nodes between high degree nodes (nodes with a degree greater than a
+given threshold) and low degree neighbors.  Compressor nodes summarize multiple
+edges of the same type to high-degree nodes.  The implementation currently
+supports one edge type.  Dedensification also has the added benefit of reducing
+the number of edges around high-degree nodes.
+"""
+import networkx as nx
+
+__all__ = [
+    "dedensify",
+]
+
+
+def dedensify(G, threshold=2, expansive=False, prefix=None, inplace=False):
+    """
+    Reduce the number of edges around high-degree nodes in a graph
+
+    Reduces the number of edges in a graph by adding compressor nodes between
+    high degree nodes (nodes with a degree greater than a given threshold) and
+    low degree neighbors.  Dedensification also has the added benefit of
+    reducing the number of edges around high-degree nodes.  The implementation
+    currently supports one edge type
+
+    Parameters
+    ----------
+    G: graph
+       A networkx graph
+    threshold: int, optional (default=2)
+       Minimal number of degrees to be considered a high degree node
+    expansive: bool, (default=False)
+       Indicates if dedensification should occur in cases where the number of
+       edges would not be reduced
+    prefix: str or None, optional (default: None)
+       An optional prefix for denoting compressor nodes
+    inplace: bool, optional (default: False)
+       Indicates if dedensification should be done inplace
+
+    Returns
+    -------
+    dedensified networkx graph : (graph, set)
+        2-tuple of the dedensified graph and list of compressor nodes
+
+    Notes
+    -----
+    According to the algorithm in [1]_, removes edges in a graph by
+    compressing/decompressing the neighborhoods around high degree nodes by
+    adding compressor nodes that summarize multiple edges of the same type
+    to high-degree nodes.  This implementation currently supports graphs with a
+    single edge type.
+
+    When expansiveness is used, dedensification is performed on all sets of
+    low/high degree nodes, even when it will not reduce the number of edges.
+
+    When expansiveness is not used, dedensification will only be applied when
+    it would reduce the total number of edges in the given graph.
+
+    Examples
+    --------
+    Non-expansive dedensification will only add compressor nodes when doing so
+    would result in fewer edges.  Nonexpansive is the default for
+    ```expansive``::
+
+        >>> import networkx as nx
```

Comment:
Please remove all the
```
>>> import networkx as nx
```

That is always assumed in our docstring examples.

Reply:
sorry.  Done